src/resmom/stage_func.c

```diff
@@ -169,22 +169,21 @@ is_child_path(char *dir, char *path)
 
 	dir_real = malloc(sizeof(char) * (MAXPATHLEN + 1));
 	fullpath_real = malloc(sizeof(char) * (2 * MAXPATHLEN + 2));
+	if (dir_real == NULL || fullpath_real == NULL)
+		goto error_exit;
```

Comment:
There is no way for the caller to determine if something went wrong since return_value will always be TRUE or FALSE. This looks like a case where we want to return...
1 = path is a child
0 = path is not a child
-1 = error encountered

The function is only called in a few places, so it shouldn't be too difficult to update the callers.